core/org.wso2.carbon.ui/src/main/java/org/wso2/carbon/ui/internal/CarbonUIServiceComponent.java

```diff
@@ -131,6 +131,9 @@ protected void activate(ComponentContext ctxt) {
         try {
             start(ctxt.getBundleContext());
             String webContextRoot = serverConfiguration.getFirstProperty("WebContextRoot");
+            if (webContextRoot == null || webContextRoot.isEmpty()) {
+                throw new RuntimeException("WebContextRoot can't be null or empty");
```

Comment:
You can call an illegalStateException rather than the very basic RuntimeException here. And also make the description descriptive enough for the user to understand that required webContextRoot element not being present or being empty in carbon.xml is the cause and guide him to put a valid value.